rust/lance-encoding/src/encoder.rs

```diff
@@ -479,6 +485,28 @@ impl CoreArrayEncodingStrategy {
         }
     }
 
+    fn choose_encoder_for_numeric(
+        arrays: &[ArrayRef],
+        data_type: &DataType,
+        version: LanceFileVersion,
+        field_meta: Option<&HashMap<String, String>>,
+    ) -> Box<dyn ArrayEncoder> {
+        let mut encoder: Option<Box<dyn ArrayEncoder>> = None;
+
+        if version >= LanceFileVersion::V2_1 {
+            if arrays[0].data_type() == data_type {
```

Comment:
@westonpace 

> I definitely think we should bitpack string offsets

If this is desirable, maybe the simplest change is to remove the condition `if arrays[0].data_type() == data_type` here so that bit packing would be applied. But this condition seems to be added intentionally and I am not sure if this should be simply removed or not.